src/main/java/org/openstreetmap/atlas/geography/atlas/pbf/OsmPbfLoader.java

```diff
@@ -24,11 +28,34 @@
  */
 public class OsmPbfLoader
 {
+    /**
+     * {@link Closeable} version of an {@link OsmosisReader} that prevents {@link InputStream}
+     * leaks.
+     *
+     * @author matthieun
+     */
+    public static class CloseableOsmosisReader extends OsmosisReader implements Closeable
+    {
+        private final InputStream input;
+
+        public CloseableOsmosisReader(final InputStream input)
+        {
+            super(input);
+            this.input = input;
+        }
+
+        @Override
+        public void close() throws IOException
+        {
+            this.input.close();
+        }
+    }
+
     private static final Logger logger = LoggerFactory.getLogger(OsmPbfLoader.class);
     private final PackedAtlasBuilder builder = new PackedAtlasBuilder();
     private final OsmPbfProcessor processor;
-    private final Supplier<OsmosisReader> osmosisReaderSupplier;
-    private OsmosisReader reader;
+    private final Supplier<CloseableOsmosisReader> osmosisReaderSupplier;
+    private CloseableOsmosisReader reader;
```

Comment:
is there a reason why reader is a class level variable?  If ```makeNewReader``` returns a ```CloseableOsmosisReader```, then we can use try-with-resources in the ```read()``` method and leverage the ```Closeable``` interface! I believe that will also remove the need of the ```closeReader()``` method.

Reply:
I agree! This would be nice. I just wanted to do minimal changes to fix the bug, since #17 should replace this code all together at some point.